src/main/java/org/simplify4u/plugins/utils/PGPSignatureUtils.java

```diff
@@ -40,6 +41,38 @@ private PGPSignatureUtils() {
         // No need to instantiate utility class.
     }
 
+    /**
+     * Check PGP signature for bad algorithms.
+     *
+     * @param signature PGP signature instance
+     * @return Returns null if no bad algorithms used, or algorithm name if used.
+     */
+    public static String checkWeakHashAlgorithm(PGPSignature signature) {
```

Comment:
for it is easy to write unit test with mocked signature

Reply:
ah, *that* was the reason why I had not created the PR yet. The unit test was missing :-P
I've been making changes for 4 or 5 different goals, so got things mixed up.

You're right of course. I'll write a few unit tests.